### PR TITLE
Add health check for AWS keys deleted from AWS

### PR DESCRIPTION
**Pull request by ddigulescu**
Addresses #513

### CODE REVIEW

File: src/lib/aws-keys-healthcheck.js [hunk 1]
```diff
@@ -61,10 +65,17 @@ function checkAwsKeys () {
...
+			if (err && err.name === 'InvalidClientTokenId') {
+				const keyName = findKeyName(keyPair.accessKey);
+				deletedKeys = [...deletedKeys, keyName];
+			}
 			if (!err) {
 				if (data && data.AccessKeyMetadata) {
 					data.AccessKeyMetadata.forEach(keyMetadata => {
-						if (keyMetadata.Status === 'Active'/* maybe remove */ && new Date().getTime() - new Date(keyMetadata.CreateDate).getTime() > 90 * 24 * 60 * 60 * 1000) {
+						if (
+							keyMetadata.Status === 'Active' /* maybe remove */ &&
```
[the remark this answers is not in the document]
Comment (ddigulescu) [reply src/lib/aws-keys-healthcheck.js]: 👍

File: src/lib/aws-keys-healthcheck.js [hunk 2]
```diff
@@ -91,12 +101,13 @@ function inUse () {
 			severity: 3,
 			technicalSummary: 'AWS keys should be rotated after 90 days',
 			checkOutput: !inUseExpiredKey ? '' : `IAM users with expired keys: ${inUseExpiredKeyUsers.join(', ')}`,
-			panicGuide: 'Guide about how to rotate AWS keys: https://docs.google.com/document/d/1bILX3O37XmhKOtpWvox9BeZ6RW4-aOn9VzmNqc16BcQ/edit'
+			panicGuide:
+				'Guide about how to rotate AWS keys: https://docs.google.com/document/d/1bILX3O37XmhKOtpWvox9BeZ6RW4-aOn9VzmNqc16BcQ/edit'
```
Comment (simonplend) [inline src/lib/aws-keys-healthcheck.js]: Unintentional new line?

File: src/lib/aws-keys-healthcheck.js [hunk 3]
```diff
@@ -80,8 +91,7 @@ function checkAwsKeys () {
 	});
 }
 
-
-function inUse () {
+function inUse() {
```
Comment (simonplend) [inline src/lib/aws-keys-healthcheck.js]: Unintentional removal of space?

File: src/lib/aws-keys-healthcheck.js [hunk 4]
```diff
@@ -105,20 +116,35 @@ function notInUse () {
 			lastUpdated,
 			severity: 3,
 			technicalSummary: 'AWS keys should be rotated after 90 days',
-			checkOutput: !notInUseExpiredKey ? '' : `IAM users with expired keys: ${notInUseExpiredKeyUsers.join(', ')}`,
-			panicGuide: 'Guide about how to rotate AWS keys: https://docs.google.com/document/d/1bILX3O37XmhKOtpWvox9BeZ6RW4-aOn9VzmNqc16BcQ/edit'
+			checkOutput: !notInUseExpiredKey
+				? ''
+				: `IAM users with expired keys: ${notInUseExpiredKeyUsers.join(', ')}`,
+			panicGuide:
+				'Guide about how to rotate AWS keys: https://docs.google.com/document/d/1bILX3O37XmhKOtpWvox9BeZ6RW4-aOn9VzmNqc16BcQ/edit'
```
Comment (simonplend) [inline src/lib/aws-keys-healthcheck.js]: Unintentional new line? This block looked fine as it was before.

File: src/lib/aws-keys-healthcheck.js [hunk 5]
```diff
@@ -105,20 +116,35 @@ function notInUse () {
...
+	return {
+		getStatus: () => ({
+			name: 'All AWS keys are still active',
+			ok: deletedKeys.length === 0,
+			businessImpact: 'There are keys deleted from AWS',
+			lastUpdated,
+			severity: 3,
+			technicalSummary: 'AWS keys deleted from AWS should be removed from application',
+			checkOutput:
+				deletedKeys.length === 0 ? '' : `The following keys were deleted from AWS: ${deletedKeys.join(', ')}`,
```
Comment (simonplend) [inline src/lib/aws-keys-healthcheck.js]: Unintentional new line?

Comment (ddigulescu) [reply src/lib/aws-keys-healthcheck.js]: Oups, sorry, it was prettier :)

**Comment by simonplend** [comment]
:raised_hands:

**Pull request merged**